integration-tests/tests/base_test.go

```diff
@@ -36,16 +38,39 @@ import (
 
 func init() {
 	c := &check.C{}
+
+	waitForRegularPartition(c)
+
+	disableAutopilot(c)
+
+	copySnapd(c)
+}
+
+func waitForRegularPartition(c *check.C) {
 	// Workaround for bug https://bugs.launchpad.net/snappy/+bug/1498293
 	// TODO remove once the bug is fixed
 	// originally added by elopio - 2015-09-30 to the rollback test, moved
 	// here by --fgimenez - 2015-10-15
 	wait.ForFunction(c, "regular", partition.Mode)
+}
 
+func disableAutopilot(c *check.C) {
 	cli.ExecCommand(c, "sudo", "systemctl", "stop", "snappy-autopilot.timer")
 	cli.ExecCommand(c, "sudo", "systemctl", "disable", "snappy-autopilot.timer")
 }
 
+func copySnapd(c *check.C) {
```

Comment:
this (remounting rw) won't work on a squashfs system. Maybe instead stop the service, and use `systemd-activate` to launch the new `snapd` by hand?